overlord/hookstate/ctlcmd/model.go

```diff
@@ -0,0 +1,265 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package ctlcmd
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/snapcore/snapd/asserts"
+	"github.com/snapcore/snapd/client/clientutil"
+	"github.com/snapcore/snapd/i18n"
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/overlord/assertstate"
+	"github.com/snapcore/snapd/overlord/ifacestate"
+	"github.com/snapcore/snapd/overlord/snapstate"
+	"github.com/snapcore/snapd/overlord/state"
+	"github.com/snapcore/snapd/snap"
+)
+
+var (
+	shortModelHelp = i18n.G("Get the active model for this device")
+	longModelHelp  = i18n.G(`
+The model command returns the active model assertion information for this
+device.
+
+By default, the model identification information is presented in a structured,
+yaml-like format, but this can be changed to json by using the --json flag.
+`)
+)
+
+func init() {
+	addCommand("model", shortModelHelp, longModelHelp, func() command { return &modelCommand{} })
+}
+
+type modelCommand struct {
+	baseCommand
+	Assertion bool `long:"assertion"`
+	Json      bool `long:"json"`
+}
+
+type modelCommandFormatter struct {
+	snapInfo *snap.Info
+}
+
+// GetEscapedDash implements part of the clientutil.ModelFormatter interface
+func (mf modelCommandFormatter) GetEscapedDash() string {
+	return "--"
+}
+
+// LongPublisher implements part of the clientutil.ModelFormatter interface
+func (mf modelCommandFormatter) LongPublisher(storeAccountID string) string {
+	if mf.snapInfo == nil || mf.snapInfo.Publisher.DisplayName == "" {
+		return mf.GetEscapedDash() + "\033[0m"
```

Comment:
Ah thanks for catching this! I'll remove the color off ascii code as no colors should appear in this context - so actually it's completely unneeded. I'll also make a comment that it should match whether happens in colors.go. It doesn't make sense IMHO to make something shared, because the output generated by this function is either YAML or JSON that will probably will not be consumed by humans, so the output is rather bare and only uses very minor things from color.go.

But this is just my opinion!